code/client/installhelper

```diff
@@ -0,0 +1,178 @@
+#!/usr/local/munki/munki-python
+# encoding: utf-8
+#
+# Copyright 2011-2024 Greg Neagle.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+installhelper
+
+Created 2024-02-20.
+
+A helper tool for handling launchd items etc when installing Munki
+
+"""
+from __future__ import absolute_import
+
+# standard libs
+import glob
+import subprocess
+import time
+import sys
+
+# our libs
+from munkilib import munkilog
+from munkilib import osutils
+from munkilib import prefs
+from munkilib import FoundationPlist
+
+
+# PyLint cannot properly find names inside Cocoa libraries, so issues bogus
+# No name 'Foo' in module 'Bar' warnings. Disable them.
+# pylint: disable=E0401
+from SystemConfiguration import SCDynamicStoreCopyConsoleUser
+# pylint: enable=E0401
+
+
+CHECK_TIME = 10
+LAUNCHD_PREFIX = 'com.googlecode.munki.'
+PROCESS_ID = 'com.googlecode.munki.installhelper'
+PROG_NAME = 'managedsoftwareupdate.py'
+
+def log(msg):
+    '''Logs messages from this tool with an identifier'''
+
+    munkilog.log('%s: %s' % (PROCESS_ID, msg))
+
+def is_managedsoftwareupdate_py_running():
```

Comment:
I'll note that in Munki 6.3 and later, the Python script name is actually ".managedsoftwareupdate.py".
You'll also need to handle the case of upgrading a pre-6.3 install where the name of the script that is running is "managedsoftwareupdate" (no ".py" extension, and no leading ".")

Reply:
Thanks @gregneagle.

Amended the function name and it's doc string.

Looking at: https://github.com/munki/munki/blob/main/code/client/munkilib/osutils.py#L156-L189, it appears that checking for `managedsoftwareupdate` will be sufficient as the ps command gives output such as:

```
90515 /usr/local/munki/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python /usr/local/munki/.managedsoftwareupdate.py
```

So the above would match as includes `MacOS/Python` and  `managedsoftwareupdate`, and the prior releases should match too.